### ISSUE TITLE
[AD] SCORPIO-Interface: Skip the F90 bridge and call the C version of SCORPIO directly

### ISSUE THREAD

**Issue by AaronDonahue** [E3SM-Project/scream#792]
Currently the SCORPIO interface follows the EAM paradigm of using F90 code to call SCORPIO F90 code, which is really a bridge to SCORPIO C code.  Because SCREAM is written in C++ this means we,

Call C++ code, which bridges to F90 code, which calls SCORPIO-F90 code, which bridges to SCORPIO-C code.

This task is to cut out the middle man of F90 code and just go from C++ SCREAM to C SCORPIO.

This task involves a number of sub-tasks and can be broken into bite sized pieces.  The SCREAM to SCORPIO interface described above is made up of a number of individual routines.  Each of which could be ported independently.

NOTE:  This task *does* provide an opportunity to rethink the SCREAM->SCORPIO interface to fit a paradigm where there is no C++ code.  For example, currently, a lot of persistent information is stored in an F90 module.  Removal of F90 means we can redesign how this persistent information is handled, and do so in a C++ way.

In other words, this task may be best served by a pre-planning stage before any implementation is executed.

**Comment by AaronDonahue** [comment]
Perhaps orthogonal to this task, but could be taken on at the same time.  Would it be possible (or beneficial) to use views of packs rather than views of Reals in the data management infrastructure that eventually calls scorpio routines?  I'm thinking scorpio_output and scorpio_input...  @bartgol , @PeterCaldwell

**Comment by bartgol** [comment]
I don't think you can. As far as I know, scorpio does not support strided/padded arrays. So we cannot store a view of packs, since it would contain garbage.

**Comment by AaronDonahue** [comment]
Right, I was thinking for the interface but then converting to Real arrays inside the interface before passing to PIO.

**Comment by AaronDonahue** [comment]
using `scalarize` for example.

**Comment by bartgol** [comment]
`ekat::scalarize` does not remove the padding, so it would not work. Creating temporaries on the fly in the c2f routines is not great, if the I/O has high frequency.

**Comment by AaronDonahue** [comment]
ahhh Okay, that makes sense. Thanks for chiming in
